v2/pkg/engine/resolve/resolve.go

```diff
@@ -333,27 +334,47 @@ func (r *Resolver) handleEvent(event subscriptionEvent) {
 		r.handleTriggerUpdate(event.triggerID, event.data)
 	case subscriptionEventKindTriggerDone:
 		r.handleTriggerDone(event.triggerID)
+	case subscriptionEventKindTriggerInitialized:
+		r.handleTriggerInit(event.triggerID)
 	case subscriptionEventKindUnknown:
 		panic("unknown event")
 	}
 }
 
+func (r *Resolver) handleTriggerInit(triggerID uint64) {
+	trig, ok := r.triggers[triggerID]
+	if !ok {
+		return
+	}
+	trig.initialized = true
+
+	if r.reporter != nil {
+		r.reporter.SubscriptionCountInc(1)
+		r.reporter.TriggerCountInc(1)
+	}
+}
+
 func (r *Resolver) handleTriggerDone(triggerID uint64) {
 	trig, ok := r.triggers[triggerID]
 	if !ok {
 		return
 	}
-	delete(r.triggers, triggerID)
+	isInitialized := trig.initialized
 	wg := trig.inFlight
 	subscriptionCount := len(trig.subscriptions)
+
+	trig.cancel()
+
+	delete(r.triggers, triggerID)
+
 	go func() {
 		if wg != nil {
 			wg.Wait()
 		}
 		for _, s := range trig.subscriptions {
 			s.writer.Complete()
 		}
-		if r.reporter != nil {
+		if r.reporter != nil && isInitialized {
```

Comment:
I don't think you can access a shared value "isInitialized" from a goroutine

Reply:
It is not shared. It is copied before to `isInitialized` the same we do for the subscription count.